### DISCUSSION TITLE
Is it possible to set flowrate and linewidth of each line type separately?

### DISCUSSION THREAD

**Discussion by jarnMod** [supermerill/SuperSlicer#1779] [Q&A] [answered]
Hi there
I'm a user of Cura Arachne and I've been testing Super Slicer in case I'd switch over. SS load faster and has a better UI, so I really want to switch over entirely. However, there are some features I wish SS could offer. One of them is an ability to set flow and line width separately. I don't know if there is a way right now, so I'm here to ask. If there is no way right now, I'd like to suggest this to be a feature.

Also, is it possible to set spacing and linewidth separately? Right now the two are linked and cannot be changed independently.

To explain this functionality so the community knows what I'm talking about, I have to compare it to Cura where the "Quality" tab has linewidth parameters (like SS) and the "Material" tab has flowrate parameters. The linewidth is the distance between lines. Changing this also increase flowrate. The flowrate setting in material chances the flowrate, the amount of plastic extruded, but not the distance between lines. This functionality enables me to do these:

1. Improve my bed adhsion: My bed is warped in a weird way. It's like a turtle shell, so I can't really put paper in the middle to improve it. Whatever I do, it will still make the middle of the bed too close. If I set it so the middle is perfect, the rim will be too far.

This functionality allows me to add more linewidth at 0.48 (equivalent to flowrate 120%) but set flowrate to 80% which allows the line to be far away from each other without getting overextrude. It's similar to 96% flow but lines are far apart, providing good adhesion despite imperfect bed. No need for ABL. No need to buy a new bed.

2. Fine tune outer wall fullness. I've a lot of model that if I adjust linewidth, it will lose some small detail on wall. If linewidth is too low, there will be some underextrusion. If the linewidth is too big, some details are lost. Ability to increase just the flow allows me to maintain the detail with the desire fullness of the wall. 

3. Combat severe moist filament. There was that one silky filament that has moisture pop all the time. It makes a lot of holes on the print part. Drying has improve the case, but given the environment where I am, I cannot rely on that. I have, however, printed successfully by using a combination of the functionality to set flow and linewidth separately and slow speed. What I did is I set linewidth to thin, like 0.32, only for the inner wall and outer wall. Increase overlap. Then ramp flowrate up to 150% for inner wall but not on the outer wall. Slow print speed to 30. What this does is when there is a pop, a hole will appear on the wall. However, highflow and close linewidth result in horizontal oozing of the filament that plug the hole. Excess flow where there is no pop tends to cause over extrusion inside the part, which is no concern. This is something I didn't even expect it to work, but it does as shown below. 
![image](https://user-images.githubusercontent.com/50398948/140058248-bd57cf5a-9d27-4285-9bb4-b31608d16a1d.png)
Before I use this kind of setting, the part is unprintable. I find some of the test cubes. These are printed at 103 flow. The top on is 103 on all lines. Line width was 0.4. The bottom one is printed at the exact same setting except the inner wall is 110 flow. You can see that some of the underextrusion are there but it's not as deep. This is what I mean by the inner wall oozed out into the gap. I tried setting the outer wall at more flow and larger linewidth, it's not as good as doing it this way. It's not gonna fix it entirely, but it saved the poor roll from going into a trash can. 
![image](https://user-images.githubusercontent.com/50398948/140060063-4e7bc1f5-7e68-48a3-8338-33e247d00b3d.png)

These are 3 use cases I've used the ability to set Line Width and Flow Rate separately. Therefore, It'd be a great boon for Super Slicer to offer something similar. I've seen that there are a few options that can provide similar effect on the 1st layer, but I've not seen the ability to do that on other layers. If there are, please tell. 

Actually, if the spacing and linewidth can be set separately, that'd be something similar, I think, except then the overlap will become an issue that may have to be addressed later.

**Comment by supermerill** [comment, marked as answer]
1- Improve my bed adhesion

You have first layer with & first layer flow rate. I don't know what you miss.

2- increasing flow of external perimeter without the width

Not supported. you can hack it by using a macro with M221 in the 'feature change custom gcode'
something like `{if next_feature == 'external_perimeter'}M221 S120{else}M221 S100{endif}`

3 - same as 2

same as 2

**Reply by jarnMod** [reply]
That macro seems to be useful. I'm gonna try that out.

**Reply by supermerill** [reply]
you can nest it if needed with {elsif }, see https://www.help.prusa3d.com/en/article/macros_1775
For the variables available, you can look at the tooltip of the field and at the wiki here.
